codepipeline-stack.yaml

```diff
@@ -0,0 +1,543 @@
+Description: One Observability Workshop Pipeline
+Parameters:
+  EnvironmentName:
+    Description: An environment name that is prefixed to resource names
+    Type: String
+    Default: OneObservability
+
+  VpcCIDR:
+    Description: Please enter the IP range (CIDR notation) for this VPC
+    Type: String
+    Default: 10.192.0.0/16
+
+  PublicSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the public subnet in the first Availability Zone
+    Type: String
+    Default: 10.192.10.0/24
+
+  PrivateSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the private subnet in the first Availability Zone
+    Type: String
+    Default: 10.192.20.0/24
+
+  UserRoleArn:
+    Description: "ARN of the Role that will have access to manage the EKS Cluster"
+    Type: String
+
+  GithubBranch:
+    Description: "Source branch to use for CodePipeline deployment"
+    Type: String
+    Default: main
+
+Resources:
+
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VpcCIDR
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: !Ref EnvironmentName
+
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: !Ref EnvironmentName
+
+  InternetGatewayAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      InternetGatewayId: !Ref InternetGateway
+      VpcId: !Ref VPC
+
+  PublicSubnet1:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Subnet - Public IP : Ensure Amazon Virtual Private Cloud (Amazon VPC) subnets are assigned a public IP address. You enable instances launched in those subnets to have a publicly accessible IP. This allows them to communicate with the internet directly, making it easier to access resources hosted within the VPC from external networks. However, note that enabling public IP assignment also exposes the instances to potential security risks, so make sure to implement appropriate security measures like network access control lists (ACLs) and security groups